src/pages/index.js

```diff
@@ -0,0 +1,382 @@
+import {
+  enableValidation,
+  validationConfig,
+  resetValidation,
+} from "../scripts/validation.js";
+import "../pages/index.css";
+import Api from "../utils/Api.js";
+import { setButtonText } from "../utils/helpers.js";
+import { disableButton } from '../scripts/validation.js';
+
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c1da3710-50e8-4d77-9af1-7ab507c7d59c",
+    "Content-Type": "application/json",
+  },
+});
+
+//destrucure the second item in the callback of the .then()
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    console.log(cards);
+
+    // render card
+    cards.forEach((card) => renderCard(card, "append"));
+    // initialCards.forEach((item) => renderCard(item, "append"));
+
+    const avatarElement = document.querySelector(".profile__avatar");
+    const nameElement = document.querySelector(".profile__name");
+    const aboutElement = document.querySelector(".profile__about");
+
+    if (userInfo) {
+      avatarElement.src = userInfo.avatar; // Set avatar image
+      nameElement.textContent = userInfo.name; // Set user's name
+      aboutElement.textContent = userInfo.about; // Set user's bio/description
+    }
+  })
+
+  .catch((err) => {
+    console.log("error fetching data:", err);
+  });
+
+//pass settings object to the vailidation functions that are called on this file
+
+// Profile Elements
+const profileEditButton = document.querySelector(".profile__edit-button");
+const cardModalButton = document.querySelector(".profile__add-button");
+const avatarModalButton = document.querySelector(".profile__avatar-button");
+const profileDescription = document.querySelector(".profile__description");
+const profileName = document.querySelector(".profile__name");
+
+//Form Elements
+const editModal = document.querySelector("#edit-profile-modal");
+const editFormElement = editModal.querySelector(".modal__form");
+const editProfileModal = document.querySelector("#edit-profile-modal");
+const closeButtons = document.querySelectorAll(".modal__close");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitButton = cardModal.querySelector(".modal__button");
+const cardModalCloseButton = cardModal.querySelector(".modal__close");
+const descriptionInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+const settings = {
+  inactiveButtonClass: "button--disabled", // Ensure this class exists in your CSS
```

Comment:
Please, delete it


You have `validationConfig` for that, with `inactiveButtonClass` inside it